File: app/src/main/java/me/srikavin/quiz/model/QuizAnswer.java
package me.srikavin.quiz.model;

import android.media.Image;

import com.google.gson.annotations.Expose;

import java.util.Objects;

import androidx.annotation.Nullable;

public class QuizAnswer {
    @Expose
    public String id;

    @Nullable
    @Expose
    public Image image;

    @Expose
    public String text;

    @Expose
    public boolean correct;
    public QuizAnswer() {
        this.text = "";
        this.correct = false;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        QuizAnswer that = (QuizAnswer) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: app/src/main/java/me/srikavin/quiz/model/QuizAnswer.kt
package me.srikavin.quiz.model

import android.media.Image

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import me.srikavin.quiz.network.common.model.data.QuizAnswerModel
import me.srikavin.quiz.network.common.model.data.ResourceId
import java.util.*

/**
 * A model class representing a quiz answer
 */
data class QuizAnswer(
        /**
         * A unique identifier used to represent a specific quiz answer
         */
        @Expose override var id: ResourceId = ResourceId(UUID.randomUUID().toString()),
        /**
         * The text contained withing the answer
         */
        @Expose @SerializedName("text") override var contents: String = "",
        /**
         * Whether or not the answer is correct
         */
        @Expose @SerializedName("correct") override var isCorrect: Boolean = false,
        @Expose internal var image: Image? = null
) : QuizAnswerModel
